File: Array&String/Python/matrix_diagonal_sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        ans, odd = 0, 1 & len(mat)
        mid = len(mat) // 2 if 1 & len(mat) else 0
        for i, sub in enumerate(mat):
            if odd and i == mid:
                ans += sub[i]
            else:
                ans += sub[i] + sub[-(i+1)]
        return ans

File: Array&String/CPP/matrix_diagonal_sum.cpp
#include <vector>
using namespace std;

class Solution {
    public:
        int diagonalSum(vector<vector<int>>& mat) {
            int n = mat.size(), ans = 0;
            int odd = n & 1, mid = n / 2;
            for (int i = 0; i < n; i++) {
                if (odd && i == mid) ans += mat[i][i];
                else ans += mat[i][i] + mat[i][n-i-1];
            }
            return ans;
        }
    };